pandas/core/missing.py

```diff
@@ -621,3 +622,24 @@ def fill_zeros(result, x, y, name, fill):
             result = result.reshape(shape)
 
     return result
+
+
+def maybe_fill(obj, value, errors):
```

Comment:
@jreback I agree with you about the function signature, will do.

The trouble is that we need some mechanism for telling the underlying function (`fillna`) to short-circuit. So we would need e.g.

```python
        try:
            missing.validate_fill_value(self, value)
        except ValueError:
            if errors == 'ignore':
                return
```

I don't know of any way of implementing that in a separate function without passing a flag somewhere. If we want to wrap this whole *sequence* of actions in one function call, that would have to be part of the method signature.

But, looking at this again, I think it'd actually be best to just keep this logic in `fillna` for now, and figure out to generalize it more broadly (so we don't have to rewrite this same bit in `shift` et. al.) in a different PR later.